lnd_methods/offchain/get_pending_payments.js

```diff
@@ -0,0 +1,190 @@
+          fee_mtokens: <Route Fee Millitokens String>
+          hops: [{
+            channel: <Standard Format Channel Id String>
+            channel_capacity: <Channel Capacity Tokens Number>
+            fee: <Fee Number>
+            fee_mtokens: <Fee Millitokens String>
+            forward: <Forward Tokens Number>
+            forward_mtokens: <Forward Millitokens String>
+            [public_key]: <Forward Edge Public Key Hex String>
+            [timeout]: <Timeout Block Height Number>
+          }]
+          mtokens: <Total Fee-Inclusive Millitokens String>
+          [payment]: <Payment Identifier Hex String>
+          timeout: <Timeout Block Height Number>
+          tokens: <Total Fee-Inclusive Tokens Number>
+          [total_mtokens]: <Total Millitokens String>
+        }
+      }]
+      created_at: <Payment at ISO-8601 Date String>
+      [destination]: <Destination Node Public Key Hex String>
+      id: <Payment Preimage Hash String>
+      [index]: <Payment Add Index Number>
+      is_confirmed: <Payment is Confirmed Bool>
+      is_outgoing: <Transaction Is Outgoing Bool>
+      mtokens: <Millitokens Attempted to Pay to Destination String>
+      [request]: <BOLT 11 Payment Request String>
+      safe_tokens: <Payment Tokens Attempted to Pay Rounded Up Number>
+      tokens: <Rounded Down Tokens Attempted to Pay to Destination Number>
+    }]
+    [next]: <Next Opaque Paging Token String>
+  }
+*/
+module.exports = ({limit, lnd, token}, cbk) => {
+  return new Promise((resolve, reject) => {
+    return asyncAuto({
+      // Check arguments
+      validate: cbk => {
+        if (!!limit && !!token) {
+          return cbk([400, 'ExpectedNoLimitWhenPagingPayFailuresWithToken']);
+        }
+
+        if (!isLnd({lnd, method, type})) {
+          return cbk([400, 'ExpectedLndForGetPendingPaymentsRequest']);
+        }
+
+        return cbk();
+      },
+
+      // Get all payments
+      listPayments: ['validate', ({}, cbk) => {
+        let offset;
+        let resultsLimit = limit || defaultLimit;
+
+        if (!!token) {
+          try {
+            const pagingToken = parse(token);
+
+            offset = pagingToken.offset;
+            resultsLimit = pagingToken.limit;
+          } catch (err) {
+            return cbk([400, 'ExpectedValidPagingTokenForGetPending', {err}]);
+          }
+        }
+
+        return lnd[type][method]({
+          include_incomplete: true,
+          index_offset: offset || Number(),
+          max_payments: resultsLimit,
+          reversed: true,
+        },
+        (err, res) => {
+          if (!!err) {
+            return cbk([503, 'UnexpectedGetPendingPaymentsError', {err}]);
+          }
+
+          if (!res || !isArray(res.payments)) {
+            return cbk([503, 'ExpectedPendingPaymentsInListPaymentsResponse']);
+          }
+
+          if (typeof res.last_index_offset !== 'string') {
+            return cbk([503, 'ExpectedLastIndexOffsetWhenRequestingFailed']);
```

Comment:
this error code should reference pending payments rather than failed

Reply:
fixed https://github.com/alexbosworth/lightning/pull/63/commits/779cea3a512c4c207530cbc16eff042604807c45